### PR TITLE
doc: Add relay basics

### LINKED ISSUE #544: Add learning material for LoRaWAN relay basics
#### Summary

Add learning material for LoRaWAN relay basics. 

#### Why do we need this ?

Still we don't have any learning materials for relays.

#### What is already there? What do you see now?

...

#### What is missing? What do you want to see?

...

#### How do you propose to document this?

- LoRaWAN network architecture with relay 
- Relay features
- WOR and WOR-ACK frames and message flow diagrams
- Device identifiers and root keys used in relays
- Uplink message format
- A few use cases

#### Can you do this yourself and submit a Pull Request?

Yes

### PR DESCRIPTION
**Pull request by pradeeka7**
#### Summary

Add LoRaWAN relay basics.
Closes https://github.com/TheThingsNetwork/docs/issues/544

#### Screenshots

image

#### Changes

- ...
- ...

#### Notes for Reviewers

@KrishnaIyer @adriansmares @nejraselimovic 
- Please review 
- This is a basic document, and I believe we don't need to present more technical details here, as mentioned in the LoRaWAN® Relay Specification TS011-1.0.0.
- References: 
https://info.semtech.com/lorawan-relay-white-paper
https://resources.lora-alliance.org/faq/faq-lorawan-relay
https://www.youtube.com/watch?v=wydJ2BsflIQ
https://resources.lora-alliance.org/technical-specifications/ts011-1-0-0-relay
https://resources.lora-alliance.org/technical-specifications/rp002-1-0-4-regional-parameters

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [X] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+title: LoRaWAN Relay
+section: The Things Fundamentals
+weight: 50
+images:
+- relay-placement.png
+---
+
+Relays are suitable for bridging wireless communication gaps in areas where the gateway and end device cannot directly communicate with each other due to weak signal strength because of factors such as extreme distance or obstacles between the gateway and end device. They are low-cost, low-power devices. A relay's hardware is equivalent to a standard LoRaWAN end device. A single relay can serve up to 16 end devices. For the Network Server, a relay appears as a standard end device.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
LoRaWAN relays are suitable for bridging wireless communication gaps in areas where the gateway and the end device cannot directly communicate with each other due to weak signal strength caused by factors such as extreme distance or obstacles between them. They are low-cost, low-power devices. A relay's hardware is equivalent to a standard LoRaWAN end device. A single relay can serve up to 16 end devices. For the Network Server, a relay appears as a standard end device.
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+The following figure shows how a relay helps to forward/relay a message from an end device to the gateway.
+
+{{< figure src="relay-placement.png" alt="Relay Placement" >}}
+
+{{< note "Adding a new gateway is a good idea if many devices have a lack of signal reception. But if it's just a few devices, using a relay is a cheaper solution." />}}
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: remove note tags

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## Relay Requirments
+
+- End device should run the LoRaWAN stack (TS001-LoRaWAN L2 1.0.4 and Regional Parameters RP2-1.0.3 specifications) + LoRa Basics Modem firmware. The experimental release of this firmware can be found at https://github.com/Lora-net/SWL2001
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
- End device should implement the LoRaWAN stack, precisely TS001-LoRaWAN L2 1.0.4 and Regional Parameters RP2-1.0.3 specifications, and LoRa Basics Modem firmware. The experimental release of the LoRa Basics Modem firmware can be found [here](https://github.com/Lora-net/SWL2001).
```

Comment (nejraselimovic) [reply doc/content/lorawan/lorawan-relay/_index.md]: Can you pls link specifications

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+- End device should be implemented with SX1261/2, LR1110, LR1120, or LR1121 sub-GHz LoRa transceiver.
+- A Network Server supporting the relay specification.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: Which network server?

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- The gateways do not need to be updated to support relays.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: This is not a requirement, so remove it from this bullet list, you can put it below

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+{{< note "The relay specification requirements are described in the  [LoRaWAN® Relay Specification TS011-1.0.0](https://resources.lora-alliance.org/technical-specifications/ts011-1-0-0-relay) document." />}}
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: remove note tags and place this sentence on top, i.e. just below the title of this section

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+## How Relay Works?
+
+The following steps describe how an end device communicates with the Network Server through a relay.
+
+- The end device that wants to use the support of a relay, first sends a  Wake-on-Radio (WOR) frame. The WOR frame is used to wake up the relay and share the radio parameters like frequency and data rate that will be used in the subsequent uplink frame. There are two types of WOR frames an end device can send to a relay (The type of the WOR frame can be identified by examining its payload content):
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
- The end device that wants to use the support of a relay, first sends a  Wake-on-Radio (WOR) frame. The WOR frame is used to wake up the relay and share the radio parameters like frequency and data rate that will be used in the subsequent uplink frame. There are two types of WOR frames an end device can send to a relay, whose type can be identified by examining the payload content:
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    - Relay Join-Request
+    - Relay Uplink (Class A uplink)
+
+{{< note "The end device and the relay know in advance which channels (frequencies) are used to send/receive WOR/WOR-ACK frames. However, the end device is not yet aware of other radio parameters used by the relay, so it assumes the CAD periodicity is 1 second, the relay's real-time clock accuracy is 40 ppm, and the number of LoRa symbols taken to move from the CAD to the RX state is 8." />}}
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: remove note tags

Comment (nejraselimovic) [reply doc/content/lorawan/lorawan-relay/_index.md]: you're using CAD abbreviation here, then explain what it means in text below, please change that - they should be explained with the first usage

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+- The relay goes to sleep mode if there is no radio activity. Even if the relay is in sleep mode, it periodically scans the channel for any radio activity in a very low-power mode. This channel scanning period is called Channel Activity Detection (CAD). The CAD can be 25ms to 1s (default). The duration between two consecutive CAD is called CADPeriodicity. The WOR frame's preamble has a variable length and can be up to 1s long. This long preamble gives the relay a higher chance of detecting the WOR frame with its very short CAD. If there is no LoRa activity is detected during the CAD, the relay goes back to sleep mode.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
- The relay goes to sleep mode if there is no radio activity. Even if the relay is in sleep mode, it periodically scans the channel for any radio activity in a very low-power mode. This channel scanning period is called Channel Activity Detection (CAD). The CAD can be 25ms to 1s (default). The duration between two consecutive CAD is called CAD periodicity. The WOR frame's preamble has a variable length and can be up to 1s long. This long preamble gives the relay a higher chance of detecting the WOR frame with its very short CAD. If there is no LoRa activity detected during the CAD, the relay goes back to sleep mode.
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+- If the relay detects a LoRa activity during the scan, it switches from CAD to the reception mode. The time taken to switch from CAD to the reception mode is called CadToRx delay. In this mode, the relay demodulates the incoming WOR frame. If the demodulation is successful and the WOR frame is valid, the relay sends the Wake On Radio Acknowedge (WOR-ACK) frame to the end device.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
- If the relay detects a LoRa activity during the scan, it switches from CAD to the reception mode. The time taken to switch from CAD to the reception mode is called CadToRx delay. In this mode, the relay demodulates the incoming WOR frame. If the demodulation is successful and the WOR frame is valid, the relay sends the Wake On Radio Acknowledge (WOR-ACK) frame to the end device.
```

Comment (nejraselimovic) [reply doc/content/lorawan/lorawan-relay/_index.md]: acknowledge or acknowledgement?

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 7 above)
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: No need for this whole section to be a bullet list, pls remove

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+- The end device finds the following information in the payload of the WOR-ACK frame. These parameters are then used by the end device to align certain settings with the relay.
+    - Data rate between relay and Network Server
+    - Relay XTAL accuracy
+    - CAD periodicity
+    - TOffset
+    - CAD to Rx delay
+    - Forward capacity
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: Can you explain these briefly

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+- The relay listens for the subsequent uplink LoRaWAN frame from the end device. Upon receiving it, the relay prepends a 6-byte metadata field to the frame. The resulting frame is called 'Relay Uplink FRMPayload'. Subsequently, the relay creates a new LoRaWAN frame by inserting the 'Relay Uplink FRMPayload' into its payload field. This new frame is then transmitted to the Network Server through the nearby gateways (FPort = 226). The Network Server handles the message deduplication.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: Why introducing ' ' quotes here? We usually don't use them and you haven't used them for other parameters either

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+- The relay receives any downlink frame from the Network Server (FPort = 226) over RX1 or RX2 slots. Upon receiving it, the relay extracts the actual payload from the LoRaWAN frame and then sends it to the end device over the RXR slot.
+
+## Regional Parameters for Relays
+
+The following regional parameters are applicable to use with relays in different regions.
+
+In EU868, four dedicated channels with a bandwidth of 125 kHz and a spreading factor of 9 are used to communicate between the end device and the relay. Among them,
+
+Two channels are used to send the WOR frame from the end device to the relay:
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
In EU868, four dedicated channels with a bandwidth of 125 kHz and a spreading factor of 9 are used to communicate between the end device and the relay. Among them, two channels are used to send the WOR frame from the end device to the relay:
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+Two channels are used to send the WOR-ACK frame from the relay to the end device:
+
+| Channel | Frequency |
+| ------- | --------- |
+| 0 | 865.3 MHz |
+| 1 | 865.9 MHz |
+
+In US915, four dedicated channels with a bandwidth of 500 kHz and a spreading factor of 10 are used to communicate between the end device and the relay. Among them,
+
+Two channels are used to send the WOR frame from the end device to the relay:
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
In US915, four dedicated channels with a bandwidth of 500 kHz and a spreading factor of 10 are used to communicate between the end device and the relay. Among them, two channels are used to send the WOR frame from the end device to the relay:
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+
+Refer to the [Regional Parameters Specification 1.0.4](https://resources.lora-alliance.org/document/rp002-1-0-4-regional-parameters) document for channel frequencies, bandwidths, and spreading factors in other regions, such as AU915, CN470, AS923, KR920, IN865, and RU864.
+
+## Security
+
+The relay has root keys (AppKey and NwkKey) the same as those of the standard end device, which are used to derive session keys (AppSKey and NwkSKey). However, these session keys cannot be used to calculate the MIC and encrypt the payload field of WOR and WOR-ACK frames. Therefore, a new root key is defined, called the Root Relay Session Key (RootWorSKey), which is used by both the relay and the end device.
+
+The end device derives the RootWorSKey from its NwkSkey (LoRaWAN Version 1.0.X) or NwkSEncKey (LoRaWAN Version 1.1.X and higher). The relay will also receive a copy of the RootWorSKey from the Network Server.
+
+The RootWorSKey is an intermediate key and is never used directly to calculate MIC or encrypt/decrypt data. Therefore, based on the RootWorSKey and the DevAddr, two session keys are derived by both the end device and the relay:
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]: you're mentioning twice that it cannot be used to calculate MIC and encrypt/decrypt payload

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+- WOR Integrity Session Key (WorSIntKey) - used by both the end device and the relay to calculate and verify the MIC of the WOR and WOR-ACK frames.
+- WOR Encryption Session Key (WorSEncKey) - used by both the end device and the relay to encrypt/decrypt the payload field of the WOR and WOR-ACK frames.
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md]:
```suggestion
- WOR Integrity Session Key (WorSIntKey) - used by both the end device and the relay to calculate and verify the MIC of the WOR and WOR-ACK frames
- WOR Encryption Session Key (WorSEncKey) - used by both the end device and the relay to encrypt/decrypt the payload field of the WOR and WOR-ACK frames
```

File: doc/content/lorawan/lorawan-relay/_index.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
+---
+title: LoRaWAN Relay
+section: The Things Fundamentals
+weight: 50
+images:
+- relay-placement.png
```
Comment (nejraselimovic) [inline doc/content/lorawan/lorawan-relay/_index.md:6]: where is this image from?

**Review by nejraselimovic — CHANGES_REQUESTED** [review]

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 9 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md]: CADPeriodicity - without space

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 11 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md]: expalined briefly. I have updated the field names as mentioned in the spec..

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 10 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md]: Acknowledge

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 8 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md]: I have removed some parts of the above paragraph and moved the rest to the beginning of the section.

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 5 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md]: I put it in the note box.

File: doc/content/lorawan/lorawan-relay/_index.md (hunk 17 above)
Comment (pradeeka7) [reply doc/content/lorawan/lorawan-relay/_index.md:6]: https://www.thethingsindustries.com/news/the-things-industries-announces-support-for-lorawan-relay-to-enhance-iot-connectivity/
I have updated it to show how the relay can fill the communication gap.

**Review by nejraselimovic — APPROVED** [review]

**Pull request merged**